src/AWS.Deploy.Orchestration/OptionSettingHandler.cs

```diff
@@ -11,16 +12,30 @@ namespace AWS.Deploy.Orchestration
 {
     public class OptionSettingHandler : IOptionSettingHandler
     {
+        private readonly IOptionSettingValuesFactory _optionSettingValuesFactory;
+
+        public OptionSettingHandler(IOptionSettingValuesFactory optionSettingValuesFactory)
+        {
+            _optionSettingValuesFactory = optionSettingValuesFactory;
+        }
+
         /// <summary>
         /// Assigns a value to the OptionSettingItem.
         /// </summary>
         /// <exception cref="ValidationFailedException">
         /// Thrown if one or more <see cref="Validators"/> determine
         /// <paramref name="value"/> is not valid.
         /// </exception>
-        public void SetOptionSettingValue(OptionSettingItem optionSettingItem, object value)
+        public async Task SetOptionSettingValue(OptionSettingItem optionSettingItem, object value)
         {
-            optionSettingItem.SetValue(this, value);
+            List<string>? disallowedValuesList = null;
+            if (optionSettingItem.DisallowedValuesSource.HasValue &&
+                _optionSettingValuesFactory.GetDisallowedValues(optionSettingItem.DisallowedValuesSource.Value) is var disallowedValues && disallowedValues != null)
+            {
+                disallowedValuesList = await disallowedValues.GetList();
```

Comment:
Per internal discussion we had on 5/12 with @normj, sounds like we want to avoid anything that could potentially make a network call inside the `SetOptionSettingValue`->`SetValue` codepath. Which I believe `GetList` is doing here.